### PR TITLE
react todo app with api

### PR DESCRIPTION
**Pull request by Arsen-Shpak**
[DEMO LINK](https://Arsen-Shpak.github.io/react_todo-app-with-api/)

### CODE REVIEW

File: src/components/Auth/AddTodo.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+type Props = {
+  newTodoField: React.RefObject<HTMLInputElement>;
```
Comment (Anna-Borodaienko) [inline src/components/Auth/AddTodo.tsx:4]: But you don't use this ```ref``` here.
And after adding you don't focus on the input field.
Add the same UseEffect with this ```ref``` as at App.tsx

File: src/App.tsx [hunk 2]
```diff
@@ -1,223 +1,252 @@
...
+
+      return todo;
+    });
+  };
+
+  // const handleEditTodo = async (id: number, comleted: boolean) => {
+  //   setIsCompleted(comleted);
+  //   editTodo(id, comleted);
+  //   getTodosFromServer();
+  // };
```
Comment (Anna-Borodaienko) [inline src/App.tsx:111]: Delete commented parts

File: src/App.tsx [hunk 3]
```diff
@@ -1,223 +1,252 @@
...
+        {todos.length > 0 && (
+          <TodoList
+            todos={filteredTodos}
+            getTodosFromServer={getTodosFromServer}
+            isAdding={isAdding}
+            tempTodo={tempTodo}
+            setHasError={setHasError}
+            setMessageError={setMessageError}
+            setIsLoading={setIsLoading}
+            isLoading={isLoading}
```
Comment (Anna-Borodaienko) [inline src/App.tsx:230]: It's not a good practice to use state-changing methods directly, maybe you can add handle-functions for them

**Review by Anna-Borodaienko — COMMENTED** [review]
1. "if new title is the same as the old one just cancel editing" - but I see the loader
2. If I edit one todo and leave empty title, it deletes. But then all other attemps to edit todos work incorrect.
3. If all todos are completed, toggleAll button doesn't work.
4. When I change status of some todos using toggleAll button, there are not loaders on these todos.

**Comment by Arsen-Shpak** [comment]
> 1. "if new title is the same as the old one just cancel editing" - but I see the loader
> 2. If I edit one todo and leave empty title, it deletes. But then all other attemps to edit todos work incorrect.
> 3. If all todos are completed, toggleAll button doesn't work.
> 4. When I change status of some todos using toggleAll button, there are not loaders on these todos.

**Pull request closed without merging**